sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -1329,8 +1329,7 @@ private[client] class Shim_v3_0 extends Shim_v2_3 {
       isSrcLocal: Boolean): Unit = {
     val session = SparkSession.getActiveSession
     assert(session.nonEmpty)
-    val database = session.get.sessionState.catalog.getCurrentDatabase
-    val table = hive.getTable(database, tableName)
+    val table = hive.getTable(tableName)
```

Comment:
just for curiosity, why can't we use `hive.getTable(dbName, tblName)` here? It looks weird that the `loadPartition` method in `HiveShim` takes a single `tableName` parameter which is a qualified name.

Reply:
We can use. The problem is when we get the `database` here from `session.get.sessionState.catalog.getCurrentDatabase `.

Reply:
If the question is why `Shim.loadPartition` doesn't take database as an argument, yes I think we can change to take. But looks like it's to match with `Hive.loadPartition`'s signature.

Reply:
The `Hive.loadPartition` API we are using here takes a `Table` instance as a parameter, not a single table name.

Reply:
`Hive.loadPartition` for Hive-3.1.x it takes `Table`, but other Hive version such as 2.1.x takes `tablename` as `string`

Hive-3.1.0
```
  public Partition loadPartition(Path loadPath, Table tbl, Map<String, String> partSpec,
      LoadFileType loadFileType, boolean inheritTableSpecs, boolean isSkewedStoreAsSubdir,
      boolean isSrcLocal, boolean isAcidIUDoperation, boolean hasFollowingStatsTask, Long writeId,
      int stmtId, boolean isInsertOverwrite)
```

Hive-2.1.0

```
  public void loadPartition(Path loadPath, String tableName,
      Map<String, String> partSpec, boolean replace,
      boolean inheritTableSpecs, boolean isSkewedStoreAsSubdir,
      boolean isSrcLocal, boolean isAcid, boolean hasFollowingStatsTask)
```

Reply:
> If the question is why `Shim.loadPartition` doesn't take database as an argument, yes I think we can change to take. But looks like it's to match with `Hive.loadPartition`'s signature.

+1 for this ,  I will update the API signature to take the database name

Reply:
I think it's okay to don't change at least in this PR since this PR will likely be ported back. Minimised change here looks good.